server/src/lib/db/mysql.ts

```diff
@@ -42,7 +42,17 @@ export default class Mysql {
             DEFAULTS.idleTimeoutMillis,
     };
 
-    this.pool  = mysql.createPool({
+    let isEmptyString = function(str: string): boolean {
+      return typeof str !== 'undefined'
+           && !str.length;
+    };
+
+    if (isEmptyString(options.password)
```

Comment:
this is a little hard to understand what you are doing here. can we add a comment?